DOC-1969: lists and advlists plugins, text_patterns, and implicit and explicit list-creation UIs (#2693)
tinymce/tinymce-docs: modules/ROOT/pages/lists.adoc

:pluginname: Lists
:plugincode: lists

The `+lists+` plugin enables numbered and bulleted lists in {productname}. To enable advanced lists (for example, lists ordered by letter or unordered lists presented with square bullets) enable the xref:advlist.adoc[Advanced List] (`+advlist+`) plugin.

The `+lists+` plugin also normalizes list behavior between browsers. Enable it if you have problems with consistency making lists.

> It is important to note that, by default, {productname} includes both implicit and explicit list-creation user-interfaces.
> 
> Adding the `lists` plugin and the `bullist` and `numlist` toolbar items to a {productname} configuration enables the explicit list-creation user-interface (UI).
> 
> The implicit list-creation UI comes via xref:content-behavior-options.adoc#text_patterns[text patterns]. The default set of `text_patterns` enabled when {productname} is initialized include https://daringfireball.net/projects/markdown/syntax[Markdown]-based patterns that {productname} automatically turns into basic ordered and unordered lists.
> 
> If the xref:advlist.adoc[Advanced Lists] Premium plugin is part of a {productname} configuration, the default `text_patterns` set also includes patterns that provide an implicit UI for generating the advanced ordered and unordered lists this plugin makes possible.

Reviewer: While I'm not opposed to calling it an `implicit ui`, I think we can still make this easier to follow.
```suggestion
It is important to note that, by default, {productname} includes both implicit and explicit list-creation user-interfaces.

Adding the `lists` plugin and the `bullist` and `numlist` toolbar items to a {productname} configuration enables the explicit list-creation user-interface (UI).

The implicit list-creation UI comes via xref:content-behavior-options.adoc#text_patterns[text patterns], which are enabled by default in {productname} version 6. The default set of `text_patterns` enabled when {productname} is initialized include https://daringfireball.net/projects/markdown/syntax[Markdown]-based patterns that {productname} automatically create ordered and unordered lists.
```

Following that, I agree the examples should be simplified and moved to the `text_patterns` documentation. I think we can lose the first two examples and just demonstrate that with the list plugin and list `text_patterns` config but no toolbar buttons, lists can still be inserted.

== Basic lists UI: implicit and explicit

[source,js]
----
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'lists',
  toolbar: 'bullist numlist',
});
----

> == Basic lists UI: explicit lists UI but no implicit lists UI

Reviewer 2: Is there any explanation for what is meant by explicit and implicit UI? I'm not sure I understand

:pluginname: Lists
:plugincode: lists

The `+lists+` plugin enables numbered and bulleted lists in {productname}. To enable advanced lists (for example, lists ordered by letter or unordered lists presented with square bullets) enable the xref:advlist.adoc[Advanced List] (`+advlist+`) plugin.

The `+lists+` plugin also normalizes list behavior between browsers. Enable it if you have problems with consistency making lists.

It is important to note that, by default, {productname} includes both implicit and explicit list-creation user-interfaces.

Adding the `lists` plugin and the `bullist` and `numlist` toolbar items to a {productname} configuration enables the explicit list-creation user-interface (UI).

> The implicit list-creation UI comes via xref:content-behavior-options.adoc#text_patterns[text patterns]. The default set of `text_patterns` enabled when {productname} is initialised include https://daringfireball.net/projects/markdown/syntax[Markdown]-based patterns that {productname} automatically turns into basic ordered and unordered lists.

Reviewer 3: ```suggestion
The implicit list-creation UI comes via xref:content-behavior-options.adoc#text_patterns[text patterns]. The default set of `text_patterns` enabled when {productname} is initialized include https://daringfireball.net/projects/markdown/syntax[Markdown]-based patterns that {productname} automatically turns into basic ordered and unordered lists.
```

If the xref:advlist.adoc[Advanced Lists] Premium plugin is part of a {productname} configuration, the default `text_patterns` set also includes patterns that provide an implicit UI for generating the advanced ordered and unordered lists this plugin makes possible.

== Basic lists UI: implicit and explicit

[source,js]
----
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'lists',
  toolbar: 'bullist numlist',
});
----

== Basic lists UI: explicit lists UI but no implicit lists UI

[source,js]
----
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'lists',
  toolbar: 'bullist numlist',

>   `text_patterns` off.

Reviewer 3: Would be be best to change the below to `false` rather than `off`?
```suggestion
  `text_patterns` false,
```
